Pod/Classes/RealexComponent/HPPManager.swift

```diff
@@ -119,124 +114,140 @@ open class GenericHPPManager<T: Decodable>: NSObject, UIWebViewDelegate, HPPView
      * If the field is not sent in, the default language is the language that is set in your account configuration. This can be set by your account manager.
      */
-    open var language:String! = ""
+    open var language: String! = ""
 
     /**
      * Used to set what text is displayed on the payment button for card transactions. If this field is not sent in, "Pay Now" is displayed on the button by default.
      */
-    open var cardPaymentButtonText:String! = ""
+    open var cardPaymentButtonText: String! = ""
 
     /**
      * Enable card storage.
      */
-    open var cardStorageEnable:String! = ""
+    open var cardStorageEnable: String! = ""
 
     /**
      * Offer to save the card.
      */
-    open var offerSaveCard:String! = ""
+    open var offerSaveCard: String! = ""
 
     /**
      * The payer reference.
      */
-    open var payerReference:String! = ""
+    open var payerReference: String! = ""
 
     /**
      * The payment reference.
      */
-    open var paymentReference:String! = ""
+    open var paymentReference: String! = ""
 
     /**
      * Flag to indicate if the payer exists.
      */
-    open var payerExists:String! = ""
+    open var payerExists: String! = ""
 
     /**
      * Used to identify an OTB transaction.
      */
-    open var validateCardOnly:String! = ""
+    open var validateCardOnly: String! = ""
 
     /**
-	* Used to check HppRequest base64 encoding.
-	If set to true - the iOS library should decode the Base64 encoded values in the HPP request JSON
+	 * Used to check HppRequest base64 encoding.
+     *
+	 * If set to true - the iOS library should decode the Base64 encoded values in the HPP request JSON
+     *
+     * If set to false - the iOS library should just leave the values alone
+     */
+    open var isEncoded: Bool! = false
 
-	If set to false - the iOS library should just leave the values alone
+    /**
+     * Transaction level configuration to enable/disable a DCC request. (Only if the merchant is configured).
      */
-    open var isEncoded:Bool! = false
-	/**
-	* Transaction level configuration to enable/disable a DCC request. (Only if the merchant is configured).
-	*/
-	open var dccEnable:String! = ""
+    open var dccEnable: String! = ""
 
     /**
      * Supplementary data to be sent to Realex Payments. This will be returned in the HPP response.
      */
-    open var supplementaryData:Dictionary<String, String>! = [:]
+    open var supplementaryData: Dictionary<String, String>! = [:]
+
+    /**
+     * Used to add User-Agent header constant
+     *
+     * If set to true - the iOS library will add User-Agent header constant
+     *
+     * If set to false - the iOS library will't add User-Agent header constant
+     */
+    open var enableUserAgent: Bool! = false
+
+    /**
+     * Used to customize X-LCL-Request header value
+     */
+    open var lclHeaderValue: String?
```

Comment:
I should have caught this previously, but LCL is the company initials and we should not be adding anything company specific to the SDK. Let's just call this userAgentValue.

Reply:
I'll change that parameter to be a dictionary so that it could be more flexible.

Reply:
A dictionary would be more flexible but also more open ended (in other languages anyways), and we're looking at a single kvp. So is a dictionary too flexible? 

I would need to see this from the integrators perspective. I'll wait for the updated unit test.

Reply:
Yes, dictionaries are flexible too. You can check this file for example how to use it
`Example/RXPiOS/ViewController.swift`

Unfortunately we don't have unit tests for `HPPManager.swift` file. In order to add them I need to perform refactoring and extract some logic to different functions. Shall I do it now ?

Reply:
OK, so the dictionary works as I expected, and this is fine. If they want to add more they can and we wont have to touch it again for it.

Reply:
Added unit tests as well 👍